lib/urlwatch/reporters.py

```diff
@@ -175,48 +166,98 @@ def _parts(self):
             yield SafeHtml('<hr>')
 
         yield SafeHtml("""
-        <address>
-        {pkgname} {version}, {copyright}<br>
-        Website: {url}<br>
-        watched {count} URLs in {duration} seconds
-        </address>
-        </body>
-        </html>
-        """).format(pkgname=urlwatch.pkgname, version=urlwatch.__version__, copyright=urlwatch.__copyright__,
-                    url=urlwatch.__url__, count=len(self.job_states), duration=self.duration.seconds)
-
-    def _diff_to_html(self, unified_diff):
-        for line in unified_diff.splitlines():
-            if line.startswith('+'):
-                yield SafeHtml('<span class="unified_add">{line}</span>').format(line=line)
-            elif line.startswith('-'):
-                yield SafeHtml('<span class="unified_sub">{line}</span>').format(line=line)
-            else:
-                yield SafeHtml('<span class="unified_nor">{line}</span>').format(line=line)
+            <div style="font-family:monospace;font-style:italic">
+            <span>{pkgname} {version}, {copyright}</span>
+            <address>Website: <a href="{url}">{url}</a></address>
+            <span>Watched {count} URLs in {duration} seconds</span>
+            </div>
+            </body>
+            </html>""").format(pkgname=urlwatch.pkgname, version=urlwatch.__version__, copyright=urlwatch.__copyright__,
+                               url=urlwatch.__url__, count=len(self.job_states), duration=self.duration.seconds)
+
+    def _diff_to_html(self, unified_diff, is_markdown=False):
+        if is_markdown and Markdown:
+            # rebuild html from markdown using markdonw2 library's Markdown
+            markdowner = Markdown(safe_mode='escape', extras=['strike', 'target-blank-links'])
+            atags = re.compile(r'<a ')
+            htags = re.compile(r'<(/?)h\d>')
+            tags = re.compile(r'^<p>(<code>)?|(</code>)?</p>$')
+
+            def mark_to_html(text):
+                '''converts Markdown (as generated by pyhtml2text filter) back to html'''
+                if text == '* * *':  # manually expand horizontal ruler since <hr> tag is used by us to separate jobs
+                    return '-' * 80
+                pre = ''
+                if text.lstrip().startswith('* '):  # item of unordered list
+                    lstripped = text.lstrip()
```

Comment:
`str.lstrip()` by default removes all whitespace, if it's a mix of tabs and spaces, the indentation will likely be wrong, but it's probably fine, just wanted to point it out.